### PR TITLE
[Docs] Desktop Access Value and Log updates

### LINKED ISSUE #16992: Desktop Access Limitations and PNG-Based session recording
## Applies To
[Desktop Access Guides](https://goteleport.com/docs/desktop-access/introduction/)

## Details
We need to better document current limitations to Desktop Access, including PNG-based session recording.

### PR DESCRIPTION
**Pull request by alexfornuto**
Resolves #16992

- Adds a Details block covering the use of Desktop Access for critical systems, not a drop-in replacement for RDP across all use cases
- Updates the Session Recording doc for Desktop Access by
  - Removing outdated reference to recording icon
  - Adding an image for recorded sessions access in the Web UI
  - Notes log rotation for Desktop Access session recording.

On that last point I welcome feedback and input on how to better phrase.

### CODE REVIEW

File: docs/pages/desktop-access/getting-started.mdx [hunk 1]
```diff
@@ -18,6 +18,16 @@ The features documented on this page are available in Teleport `10.2.6` and high
 This guide will help you configure Teleport and Active Directory in order to
 provide secure, passwordless access to Windows desktops.
 
+<Details title="Compare Desktop Access to traditional RDP">
+
+Teleport Desktop Access is desiged to be a secure access protocol for Windows
+environments. Our protocol is designed with security as a priority, and may not
+be as performant as RDP or Guacamole. Consider Desktop Accesss to manage access
```
Comment (ptgott) [inline docs/pages/desktop-access/getting-started.mdx:25]: Desktop Access speaks RDP, so maybe say "RDP clients like Guacamole"?

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 2]
```diff
@@ -87,4 +76,8 @@ player supports toggling between play and pause, but does not support seeking to
 a specific point in the stream, rewinding, or restarting playback when the end
 of the stream is reached. To replay a session, refresh the page.
 
+## Log rotation
 
+Be aware, Desktop Access session recording comprises of raw PNG-based files.
```
Comment (ptgott) [inline docs/pages/desktop-access/reference/sessions.mdx:81]: I think we should expand "comprises of" so it's clearer how these PNG-based files will affect your infrastructure/how they travel. Will the PNG-based files be stored uncompressed on the Teleport backend, or are they stored locally to each Desktop Service instance? Is this an issue for Teleport Cloud?

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 3]
```diff
@@ -87,4 +76,8 @@ player supports toggling between play and pause, but does not support seeking to
...
+Consider the frequency and duration of Desktop Access sessions and rotate logs
```
Comment (ptgott) [inline docs/pages/desktop-access/reference/sessions.mdx:82]: Does "Consider the frequency and duration of Desktop Access sessions" mean that you might want to limit Desktop Access sessions, or that you should be tracking the length of these sessions so you can size your disk volumes correctly?

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 4]
```diff
@@ -87,4 +76,8 @@ player supports toggling between play and pause, but does not support seeking to
...
+apporpirately to avoid filling up filesystems in your infrastructure.
```
Comment (ptgott) [inline docs/pages/desktop-access/reference/sessions.mdx:83]:
```suggestion
appropriately to avoid filling up storage devices in your infrastructure.
```

File: docs/pages/desktop-access/getting-started.mdx (hunk 1 above)
Comment (zmb3) [reply docs/pages/desktop-access/getting-started.mdx:25]: Yeah, I wouldn't compare it to RDP, because it is RDP. I would compare it to more mature RDP _clients_. Teleport is not slower because of protocol, but because it is a rather simple RDP client and doesn't support advanced graphics pipelines, bitmap caching, etc.

Maybe say something like: Teleport implements a minimal feature set of the RDP protocol in order to reduce the surface area for potential attacks. This means it may not perform as well as more sophisticated RDP clients that support the full gamut of RDP features.

File: docs/pages/desktop-access/reference/sessions.mdx (hunk 3 above)
Comment (zmb3) [reply docs/pages/desktop-access/reference/sessions.mdx:82]: As mentioned offline, there's no need to consider manually rotating logs in most deployments.

File: docs/pages/desktop-access/getting-started.mdx (hunk 1 above)
Comment (alexfornuto) [reply docs/pages/desktop-access/getting-started.mdx]: I've adjusted the phrasing here.

File: docs/pages/desktop-access/reference/sessions.mdx (hunk 2 above)
Comment (alexfornuto) [reply docs/pages/desktop-access/reference/sessions.mdx]: I've updated this section based on internal discussion. Additional documentation on audit and session logs may be coming soon, which we can then link to here.

File: docs/pages/desktop-access/reference/sessions.mdx (hunk 3 above)
Comment (alexfornuto) [reply docs/pages/desktop-access/reference/sessions.mdx]: Removed this sentence.

File: docs/pages/desktop-access/reference/sessions.mdx (hunk 4 above)
Comment (alexfornuto) [reply docs/pages/desktop-access/reference/sessions.mdx]: This sentence is gone anyway.

**Review by ptgott — APPROVED** [review]

File: docs/pages/desktop-access/getting-started.mdx [hunk 5]
```diff
@@ -18,6 +18,17 @@ The features documented on this page are available in Teleport `10.2.6` and high
 This guide will help you configure Teleport and Active Directory in order to
 provide secure, passwordless access to Windows desktops.
 
+<Details title="Compare Desktop Access to traditional RDP">
```
Comment (zmb3) [inline docs/pages/desktop-access/getting-started.mdx:21]:
```suggestion
<Details title="Compare Desktop Access to other RDP clients">
```

File: docs/pages/desktop-access/getting-started.mdx [hunk 6]
```diff
@@ -18,6 +18,17 @@ The features documented on this page are available in Teleport `10.2.6` and high
...
+
+Teleport Desktop Access is desiged to be a secure access protocol for Windows
```
Comment (zmb3) [inline docs/pages/desktop-access/getting-started.mdx:23]:
```suggestion
Teleport Desktop Access is desiged to be a secure access solution for Windows
```

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 7]
```diff
@@ -87,4 +76,10 @@ player supports toggling between play and pause, but does not support seeking to
 a specific point in the stream, rewinding, or restarting playback when the end
 of the stream is reached. To replay a session, refresh the page.
 
+## Log rotation
 
+Be aware, Desktop Access session recording comprises of raw PNG-based files,
+and are stored on the host running the Desktop Access service. Session recordings
```
Comment (zmb3) [inline docs/pages/desktop-access/reference/sessions.mdx:82]: As mentioned offline, they are not stored on the desktop service long term, and in some cases are not stored on that host at all (sync recording)

I would rephrase this paragraph. Even the title of "Log rotation" is a little confusing because there are so many things in teleport that can be considered a log.

Let's say something like:

> Be aware, Desktop Access session recordings capture the entire contents of the
> screen. This means they take up significantly more disk space than SSH or Kubernetes
> session recordings. When using async recording modes, ensure that the host running
> Teleport's desktop service has sufficient disk space to store recordings that are
> in progress. If disk space is a concern, we recommend using sync recording and
> configuring a cloud storage solution such as S3.

Comment (alexfornuto) [reply docs/pages/desktop-access/reference/sessions.mdx]: I took your paragraph, but added after the line linking to the config reference.

File: docs/pages/desktop-access/getting-started.mdx (hunk 6 above)
Comment (alexfornuto) [reply docs/pages/desktop-access/getting-started.mdx]: Incorporated in https://github.com/gravitational/teleport/pull/18655/commits/7152a21cee510e763e56d80b8f8f7cf39247a1ec

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 8]
```diff
@@ -87,4 +76,14 @@ player supports toggling between play and pause, but does not support seeking to
 a specific point in the stream, rewinding, or restarting playback when the end
 of the stream is reached. To replay a session, refresh the page.
 
+## Log rotation
```
Comment (zmb3) [inline docs/pages/desktop-access/reference/sessions.mdx:79]:
```suggestion
## Storage
```

I recommend changing the title here because "log rotation" sounds like something you have to do manually, and that's not really the case here.

**Review by zmb3 — APPROVED** [review]

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 9]
```diff
@@ -87,4 +76,14 @@ player supports toggling between play and pause, but does not support seeking to
...
+
+Be aware, Desktop Access session recordings capture the entire contents of the
+screen. This means they take up significantly more disk space than SSH or Kubernetes
+session recordings. When using async recording modes, ensure that the host running
+Teleport's desktop service has sufficient disk space to store recordings that are
+in progress. If disk space is a concern, we recommend using sync recording and
+configuring a cloud storage solution such as S3.
 
+Refer to the `storage` section of the [Teleport Configuration Reference](../../reference/config.mdx)
+for more information on session logging and recoring management.
```
Comment (ibeckermayer) [inline docs/pages/desktop-access/reference/sessions.mdx:89]:
```suggestion
for more information on session logging and recording management.
```

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 10]
```diff
@@ -87,4 +76,14 @@ player supports toggling between play and pause, but does not support seeking to
 a specific point in the stream, rewinding, or restarting playback when the end
 of the stream is reached. To replay a session, refresh the page.
 
+## Storage
```
Comment (ibeckermayer) [inline docs/pages/desktop-access/reference/sessions.mdx:79]: @zmb3 should we add the results of the size estimate somewhere here?

500 bytes per 64x64 png (gzipped on disk) 
which means 250 kb per full frame change on 1080p

Comment (zmb3) [reply docs/pages/desktop-access/reference/sessions.mdx:79]: Sure! Might as well!

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 11]
```diff
@@ -87,4 +76,14 @@ player supports toggling between play and pause, but does not support seeking to
...
+
+Be aware, Desktop Access session recordings capture the entire contents of the
+screen. This means they take up significantly more disk space than SSH or Kubernetes
+session recordings. When using async recording modes, ensure that the host running
+Teleport's desktop service has sufficient disk space to store recordings that are
+in progress. If disk space is a concern, we recommend using sync recording and
+configuring a cloud storage solution such as S3.
```
Comment (ibeckermayer) [inline docs/pages/desktop-access/reference/sessions.mdx:86]:
```suggestion
Be aware, Desktop Access session recordings save PNGs of changing sections of the
screen, which means they take up significantly more disk space than SSH or Kubernetes
session recordings. When using async recording modes, ensure that the host running
Teleport's desktop service has sufficient disk space to store recordings that are
in progress. As a point of reference, when a full 1080p screen is redrawn (for example 
when opening a new full-sized application window), you can expect about 250kb to be
saved to disk. If disk space is a concern, we recommend using sync recording and
configuring a cloud storage solution such as S3.
```
